baselines/fedht/fedht/main.py

```diff
@@ -0,0 +1,163 @@
+"""Run main for fedht baseline."""
+
+import pickle
+import random
+import torch
+
+import flwr as fl
+import hydra
+import numpy as np
+from flwr.common import NDArrays, ndarrays_to_parameters
+from flwr.server.strategy.strategy import Strategy
+from flwr_datasets import FederatedDataset
+from flwr_datasets.partitioner import PathologicalPartitioner
+from omegaconf import DictConfig
+from torch.utils.data import DataLoader
+
+from fedht.client import generate_client_fn_mnist, generate_client_fn_simII
+from fedht.fedht import FedHT
+from fedht.model import LogisticRegression
+from fedht.server import fit_round, get_evaluate_fn
+from fedht.utils import MyDataset, sim_data
+
+
+@hydra.main(config_path="conf", config_name="base_mnist", version_base=None)
+def main(cfg: DictConfig):
+    """Run main file for fedht baseline.
+
+    Parameters
+    ----------
+    cfg : DictConfig
+        Config file for federated baseline; read from fedht/conf.
+    """
+    # set seed
+    random.seed(2024)
+
+    # this vs. setting in cfg; what is preferred?
+    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+    print(device)
```

Comment:
I'm not sure if it's preferred to get the device like this, or directly through the config file.

Reply:
Great question! that line should go either inside `client_fn` or in the constructor of your client. That's the preferred way because a CUDA device will be "discoverable" only if the simulation engine has been set with `num_gpus>0` (when specifying the `client_resources`). 

That being said, you can use that `device` and pass it to `get_evaluate_fn()` since the system resources (CPUs, GPUs) needed for the execution of the server & the strategy aren't managed by the Simulation Engine.